webapp/components/suggestion/at_mention_provider.jsx

```diff
@@ -126,22 +127,35 @@ function filterUsersByPrefix(users, prefix, limit, type) {
 }
 
 export default class AtMentionProvider {
-    handlePretextChanged(suggestionId, pretext) {
+    handlePretextChanged(suggestionId, pretext, argChannelId) {
```

Comment:
Ah, I thought that might be an issue. You could add a `componentWillReceiveProps` method to the `Textbox` to recreate its `suggestionProviders` whenever it's `channelId` prop changes. That way it'll use a new `AtMentionProvider` with the new `channelId` after it re-renders